qt_ui.py

```diff
@@ -0,0 +1,299 @@
+        self.errorMsg.setObjectName("errorMsg")
+        mainWindow.setCentralWidget(self.centralwidget)
+        self.menubar = QtWidgets.QMenuBar(mainWindow)
+        self.menubar.setGeometry(QtCore.QRect(0, 0, 256, 24))
+        self.menubar.setObjectName("menubar")
+        self.menuMore = QtWidgets.QMenu(self.menubar)
+        self.menuMore.setObjectName("menuMore")
+        mainWindow.setMenuBar(self.menubar)
+        self.statusbar = QtWidgets.QStatusBar(mainWindow)
+        self.statusbar.setObjectName("statusbar")
+        mainWindow.setStatusBar(self.statusbar)
+        self.actionAbout = QtWidgets.QAction(mainWindow)
+        self.actionAbout.setObjectName("actionAbout")
+        self.actionQuit = QtWidgets.QAction(mainWindow)
+        self.actionQuit.setObjectName("actionQuit")
+        self.menuMore.addAction(self.actionAbout)
+        self.menuMore.addAction(self.actionQuit)
+        self.menubar.addAction(self.menuMore.menuAction())
+        self.img = None
+        self.color = None
+        self.colHex = None
+
+        self.retranslateUi(mainWindow)
+        QtCore.QMetaObject.connectSlotsByName(mainWindow)
+
+        self.actionAbout.triggered.connect(lambda: self.functionAbout())
+        self.actionQuit.triggered.connect(lambda: self.quit)
+        self.imageSelectButton.clicked.connect(lambda: self.onImgClick())
+        self.colorSelectButton.clicked.connect(lambda: self.onColClick())
+        self.generateButton.clicked.connect(lambda: self.onGenerateClick())
+
+    def retranslateUi(self, mainWindow):
+        _translate = QtCore.QCoreApplication.translate
+        mainWindow.setWindowTitle(_translate("mainWindow", "QR Generator"))
+        self.imageSelectButton.setText(_translate("mainWindow", "Select image"))
+        self.colorSelectButton.setText(_translate("mainWindow", "Select color"))
+        self.generateButton.setText(_translate("mainWindow", "Generate"))
+        self.urlInput.setText(_translate("mainWindow", "Enter URL"))
+        self.errorMsg.setText("")
+        self.menuMore.setTitle(_translate("mainWindow", "More"))
+        self.actionAbout.setText(_translate("mainWindow", "About"))
+        self.actionQuit.setText(_translate("mainWindow", "Quit"))
+        self.actionQuit.setShortcut(_translate("mainWindow", "Ctrl+Q"))
+    
+    def functionAbout(self):
+        about = QMessageBox()
+        about.setWindowTitle("About QR Generator")
+        about.setText("QrCodeGenerator by MarcManfred\n\nUI built with PyQt5 by MiaHammer")
+        about.setIcon(QMessageBox.Information)
+        about.exec_()
+    
+    def onImgClick(self):
+        self.img = QFileDialog.getOpenFileName(filter = "Image File (*.png)")
+        if self.img[0] == '':
+            self.img = None
+        else:
+            self.img = self.img[0]
+
+    def onColClick(self):
+        self.color = QColorDialog.getColor()
+        self.generateHex()
+    
+    def onGenerateClick(self):
```

Comment:
If the user enters a file-name without an extension, the file will be generated without one. This means, that the file cannot be opened. 

I have no experience with PYQT  but from what I've found, it is not possible to let QTDialog use a default extension as you can do with e.g TKinter filedialog.
One possible solution would be as follows :

```python   
    def onGenerateClick(self):
        #inverting the condition here saves one intendation
        if not self.urlInput.text():
            self.errorMsg.setText("ERROR: no link input")
            return
        
        with concurrent.futures.ThreadPoolExecutor() as executor:
            f1 = executor.submit(self.generationThread)
            #f2 = executor.submit(self.saveFileNameThread)
            qrCode = f1.result()
            #fileName = f2.result()

            # Set a default file extension
            default_file_extension = ".png" # Probably best to lock the filter to .png

            options = QFileDialog.Options()
            options |= QFileDialog.DontUseNativeDialog  # To force the Qt dialog

            file_name = QFileDialog.getSaveFileName(None, "Save Image", "", default_file_extension,  options=options)[0]

            self.saveImage(qrCode, file_name, default_file_extension)
    
    def saveImage(self,qr_code, file_name, default_file_extension):
        if not file_name:
            self.errorMsg.setText("")
            return
        
        # If the user provided an extension, save the file.
        if os.path.splitext(file_name)[1]:
            print(file_name)
            qr_code.save(file_name)
            return
        
        # If the user didn't provide an extension, add the default one
        file_name += default_file_extension

        #since the extension was added outside of the file-dialog, we need a seperate check to check if the file exists.
        if os.path.exists(file_name):
            # Ask the user if they want to overwrite the existing file
            msg_box = QMessageBox()
            msg_box.setIcon(QMessageBox.Warning)
            msg_box.setText("File already exists. Do you want to overwrite it?")
            msg_box.setStandardButtons(QMessageBox.Yes | QMessageBox.No)
            response = msg_box.exec_()
            
            if response == QMessageBox.No:
                return

        qr_code.save(file_name)
```
        
This way, there is no need for the user to provide an extension, but if he does, his extension is used.

There are probably better ways to achieve this, but as mentioned above, I am not familiar with this module.